### PR TITLE
chore: c++17 compatibility

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -130,9 +131,44 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
...
+
+ELSE ()
+
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wshadow")
+
+ENDIF ()
 
-SET (CMAKE_CXX_STANDARD 20)
 SET (CMAKE_CXX_STANDARD_REQUIRED ON)
 SET (CMAKE_CXX_EXTENSIONS OFF)
```
Comment (kyle-cochran) [inline CMakeLists.txt:173]: Do you not need the Dockerfile changes here to add the format lib dependency for the sub-libraries?

File: CMakeLists.txt [hunk 2]
```diff
@@ -130,9 +131,44 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
...
+
+    ELSE ()
+
+        MESSAGE (STATUS "C++20 support enabled")
+
+        SET (CMAKE_CXX_STANDARD 20)
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 13.0")
```
Comment (kyle-cochran) [inline CMakeLists.txt:158]:
```suggestion
            MESSAGE (FATAL_ERROR "GCC version must be at least 13.0 to build with C++20")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -130,9 +131,44 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
...
+
+    IF (BUILD_WITH_CXX_17)
+
+        SET (CMAKE_CXX_STANDARD 17)
+
+        MESSAGE (STATUS "C++17 support enabled")
+
+        IF (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+
+            MESSAGE (FATAL_ERROR "GCC version must be at least 7.0")
```
Comment (kyle-cochran) [inline CMakeLists.txt:146]:
```suggestion
            MESSAGE (FATAL_ERROR "GCC version must be at least 7.0 to build with C++17")
```

**Review by kyle-cochran — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (vishwa2710) [reply CMakeLists.txt:173]: It was never deleted in OSTk simulation. I had missed it last time.

**Pull request merged**